pkg/perf/healthcheck/healthcheck.go

```diff
@@ -60,21 +62,35 @@ func (h *healthcheckTask) Run(ctx context.Context) (interface{}, error) {
 	log.Debug().Msg("starting health check task")
 	errs := make(map[string][]string)
 
+	cl := perf.GetZbusClient(ctx)
+	zui := stubs.NewZUIStub(cl)
+
 	for label, check := range h.checks {
-		errors := check(ctx)
-		if len(errors) == 0 {
-			continue
+		op := func() error {
+			errors := check(ctx)
+			errs[label] = errorsToStrings(errors)
+
+			if err := zui.PushErrors(ctx, label, errs[label]); err != nil {
+				return err
+			}
+
+			if len(errors) != 0 {
+				return fmt.Errorf("failed health check")
+			}
+
+			return nil
 		}
 
-		errs[label] = errorsToStrings(errors)
-	}
+		notify := func(err error, t time.Duration) {
+			log.Error().Err(err).Str("check", label).Dur("retry-in", t).Msg("failed health check. retrying")
+		}
 
-	cl := perf.GetZbusClient(ctx)
-	zui := stubs.NewZUIStub(cl)
+		bo := backoff.NewExponentialBackOff()
+		bo.InitialInterval = 30 * time.Second
+		bo.MaxInterval = 30 * time.Second
+		bo.MaxElapsedTime = 10 * time.Minute
 
-	for label := range h.checks {
-		err := zui.PushErrors(ctx, label, errs[label])
-		if err != nil {
+		if err := backoff.RetryNotify(op, bo, notify); err != nil {
 			return nil, err
```

Comment:
I don't believe you should return here, since there are other checks you need to do.